File: 2574-left-and-right-sum-differences/2574-left-and-right-sum-differences.py
class Solution:
    def leftRigthDifference(self, nums: List[int]) -> List[int]:
        leftSum = [0]
        rightSum = [0]
        leftNums = nums
        rightNums = nums
        for i in  range(len(nums) -1):
            if i == 0:
                leftSum.append(nums[i])
                rightSum.insert(0, nums[len(nums) -1]);
            else:
                leftNums = nums
                leftNums = leftNums[slice(0,i+1)]
                leftNums = sum(leftNums)
                leftSum.append(leftNums)
                rightNums = nums
                rightNums = rightNums[slice(len(rightNums) -i -1, len(rightNums))]
                rightNums = sum(rightNums)
                rightSum.insert(0,rightNums)
        for x in  range(len(nums)):
            num = leftSum[x] - rightSum[x]
            if num < 0:
                num *= -1
            nums[x] = num
        
        return nums

File: 2574-left-and-right-sum-differences/2574-left-and-right-sum-differences.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var leftRigthDifference = function(nums) {
    let leftSum = [0];
    let rightSum = [0];
    let leftNums = nums;
    let rightNums = nums;
    for(let i = 0; i < nums.length -1; i++){
        if(i == 0) {
            leftSum.push(nums[i]);
            rightSum.unshift(nums[nums.length -1]);
        }else {
            leftNums = nums;
            leftNums = leftNums.slice(0,i+1);
            leftNums = leftNums.reduce((partialSum, a) => partialSum + a, 0);
            leftSum.push(leftNums);
            rightNums = nums;
            rightNums = rightNums.slice(-i -1);
            rightNums = rightNums.reduce((partialSum, a) => partialSum + a, 0);
            rightSum.unshift(rightNums);
        }
    }
    for(let x = 0; x < nums.length; x++){
        let num = leftSum[x] - rightSum[x];
        if(num < 0) num *= -1;
        nums[x] = num;
    }
    return nums;
};